pkg/hubble/filters/patterns.go

```diff
@@ -58,6 +70,45 @@ func appendFQDNPatternRegexp(sb *strings.Builder, fqdnPattern string) error {
 	return nil
 }
 
+// appendNodeNamePatternRegexp appends the regular expression equivalent to
+// nodeNamePattern to sb. The returned regular expression matches node names
+// that include a cluster name.
+//
+// Node name patterns consist of a cluster pattern element and a node pattern
+// element separated by a forward slash. Each element is an FQDN pattern, with
+// an empty pattern matching everything. If there is no forward slash then the
+// pattern is treated as a node pattern and matches all clusters.
+func appendNodeNamePatternRegexp(sb *strings.Builder, nodeNamePattern string) error {
+	if nodeNamePattern == "" {
+		return errEmptyPattern
+	}
+	clusterPattern := ""
+	nodePattern := ""
+	elems := strings.Split(nodeNamePattern, "/")
+	switch len(elems) {
+	case 1:
+		nodePattern = elems[0]
+	case 2:
+		clusterPattern = elems[0]
+		nodePattern = elems[1]
+	default:
+		return errTooManySlashesInPattern
+	}
+
+	if clusterPattern == "" {
+		sb.WriteString(fqdnRegexpStr)
```

Comment:
Simply out of sheer curiosity (no changes request or anything): Did we ever find out if there are any restrictions actually enforced on the cluster name? 

The only reference that it must be domain-name-compatible can be found here https://docs.cilium.io/en/v1.9/gettingstarted/clustermesh/#ensure-that-the-etcd-service-names-can-be-resolved

Reply:
I think I remember reading somewhere that cluster names must be valid k8s object names. I'll try to find a reference.